client/state/happychat/middleware.js

```diff
@@ -67,6 +132,14 @@ export default function( connection = null ) {
 
 	return store => next => action => {
 		switch ( action.type ) {
+			case HAPPYCHAT_CONNECTING:
+				if ( getHappychatConnectionStatus( store.getState() ) === 'connected' ) {
+					// If chat is already connected, do nothing and stop the action from proceeding to reducers
+					return;
```

Comment:
This could be personal preference (and there may be hazards I'm not seeing) but my instinct is that when an action is dispatched through reducers, we're saying "This thing is happening now!" So in this case, if the reducers receive a `HAPPYCHAT_CONNECTING` action it's reasonable for them to assume that an attempt is being made at starting a connection.

With that in mind, if we've already made a connection then the `HAPPYCHAT_CONNECTING` option should never fire (or should at least be a noop). So the question is, where do we protect against this? Options:

1. __Prevent components from dispatching `HAPPYCHAT_CONNECTING` actions if already connected.__ This cuts it off at the source, but requires that any component wanting to offer HC connection must remember to check the connection status before dispatching the action. This feels error-prone and could have bad consequences if you forget. You'd still need safeguards in the middleware and reducers.
  
2. __Ignore `HAPPYCHAT_CONNECTING` actions in reducers.__ This is required anyways if we do option 1. But again this is error prone — right now we only have one reducer relying on `HAPPYCHAT_CONNECTING` but it's conceivable that other reducers in other parts of Calypso would be interested in this action and would also need to remember to check connection status first. This also still requires middleware protection.

3. __In middleware, prevent `HAPPYCHAT_CONNECTING` actions from heading to reducers if already connected.__ This is the only way to offer full protection in a "fool-proof"* way. Components can fire the action willy-nilly and the reducers will never see it if it isn't relevant. In my brain what we're saying is "This action shouldn't happen because it isn't possible given the current state, so we're cutting it off".

_* Note: the only possible hazard I can see with option 3 is that other middleware, if executed first, will still receive the `HAPPYCHAT_CONNECTING` action and could respond to it — but that seems highly unlikely and the only way to protect is by strict enforcement of components not firing the action if already connected. So this is an issue all options face._

Anyways — those are my (probably too-detailed) thoughts on why I implemented this way. Are there best practices or complications I'm missing? I'm not overly opinionated on this, but given my understanding of the Redux chain seemed the most protective against weird runtime bugs.

/cc @omarjackman @unDemian @jordwest